src/App/Router.php

```diff
@@ -278,12 +262,17 @@ private function getModuleClass()
 
 		$this->parameters = [];
 
-		$routeInfo  = $dispatcher->dispatch($this->httpMethod, $cmd);
+		$routeInfo  = $dispatcher->dispatch($this->args->getMethod(), $cmd);
 		if ($routeInfo[0] === Dispatcher::FOUND) {
 			$moduleClass = $routeInfo[1];
 			$this->parameters = $routeInfo[2];
 		} elseif ($routeInfo[0] === Dispatcher::METHOD_NOT_ALLOWED) {
-			throw new HTTPException\MethodNotAllowedException($this->l10n->t('Method not allowed for this module. Allowed method(s): %s', implode(', ', $routeInfo[1])));
+			if ($this->args->getMethod() === static::OPTIONS) {
```

Comment:
I still find it weird to handle the `OPTIONS` in the "method not allowed" part instead of doing it prior to this dispatcher call.

Reply:
At first I searched a "catch-all" possibility in the router library, but there is no possibility. I also found a todo near our old code that the Option method should be possible per route in the future and now that's the place where both, a catch-all and a per route option could be handled.

If we don't want a per route possibility, I can catch it before executing the router dispatching in this method, which is maybe less weird :-)

Reply:
The `OPTIONS` is simply returning the allowed HTTP methods, see here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS 

So it does make more sense to use the router configuration to fetch the assigned HTTP methods and to return them instead of having individual endpoints.